### PR TITLE
Basic Benchmark Implementation

### PR DESCRIPTION
**Pull request by mricopeng**
### Description

This pull request implements a comprehensive codebase benchmarking system designed to evaluate code understanding models. The system consists of two main phases: dataset creation and model benchmarking. Key components include:

1. Dataset Creation:
   - `CodebaseChunker`: Splits the codebase into manageable chunks
   - `QAGenerator`: Generates question-answer pairs from code chunks
   - `DatasetManager`: Handles saving and loading of datasets

2. Benchmarking:
   - `ModelTester`: Tests the model against generated QA pairs
   - `ContextEvaluator`: Evaluates context retrieval performance
   - `AnswerEvaluator`: Evaluates answer quality

The PR also introduces a YAML configuration file (`benchmark_config.yaml`) for easy customization of benchmark parameters and a README with usage instructions. The main scripts `create_dataset.py` and `run_benchmark.py` have been updated to use the new configuration system.

### Documentation

A new `README.md` file has been added, providing:
- An overview of the benchmarking tool
- Installation instructions
- Usage guidelines for dataset creation and benchmark execution
- Information on customization options

The `benchmark_config.yaml` file includes explanatory comments for each configuration field, enhancing usability.

### Changes that Break Backward Compatibility

N/A

*Created with [Palmier](https://www.palmier.io)*

### CODE REVIEW

**Comment by mricopeng** [comment]
@palmier-bot describe

File: benchmark/dataset_creation/codebase_chunker.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                'metadata': {
+                    'file': file_path,
+                    'name': node.name,
+                    'lineno': node.lineno,
+                    'end_lineno': node.end_lineno
+                }
+            }
+        else:
+            # If the chunk is too large, we need to split it
+            return self._split_large_chunk(node, file_path)
```
Comment (htin1) [inline benchmark/dataset_creation/codebase_chunker.py:88]: _split_large_chunk returns a List[Dict], should create_chunk also returns a list?

Comment (mricopeng) [reply benchmark/dataset_creation/codebase_chunker.py:88]: huh we could i guess i just parse it differently if it's a dictionary

Comment (htin1) [reply benchmark/dataset_creation/codebase_chunker.py:88]: you could parse it differently, but the same function is still returning two different types? Then in line 56:
```
chunk = self._create_chunk(node, file_path)
```

chunk could be either a List[Dict] or Dict. 

```
function_chunks.append(chunk)
```
could be appending a list to a list -> List[Dict, Dict, List[Dict], Dict...]

**Pull request closed without merging**